pkg/vulnsrc/openeuler/openeuler_test.go

```diff
@@ -0,0 +1,202 @@
+						References: []string{
+							"https://www.openeuler.org/en/security/safety-bulletin/detail.html?id=openEuler-SA-2024-1349",
+							"https://www.openeuler.org/en/security/cve/detail.html?id=CVE-2023-52604",
+							"https://nvd.nist.gov/vuln/detail/CVE-2023-52604",
+						},
+						Severity: types.SeverityHigh,
+					},
+				},
+				{
+					Key:   []string{"vulnerability-id", "openEuler-SA-2024-1349"},
+					Value: map[string]interface{}{},
+				},
+			},
+		},
+		{
+			name:    "sad path (dir doesn't exist)",
+			dir:     filepath.Join("testdata", "badPath"),
+			wantErr: "no such file or directory",
+		},
+		{
+			name:    "sad path (failed to decode)",
+			dir:     filepath.Join("testdata", "sad"),
+			wantErr: "failed to decode openEuler CVRF JSON",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			vs := NewVulnSrc()
+			vulnsrctest.TestUpdate(t, vs, vulnsrctest.TestUpdateArgs{
+				Dir:        tt.dir,
+				WantValues: tt.wantValues,
+				WantErr:    tt.wantErr,
+			})
+		})
+	}
+}
+
+func TestVulnSrc_Get(t *testing.T) {
+	tests := []struct {
+		name     string
+		fixtures []string
+		version  string
+		pkgName  string
+		dist     Distribution
+		want     []types.Advisory
+		wantErr  string
+	}{
+		{
+			name:     "happy path 1",
+			fixtures: []string{"testdata/fixtures/happy.yaml"},
+			version:  "22.03-LTS-SP2",
+			pkgName:  "perf",
+			want: []types.Advisory{
+				{
+					VulnerabilityID: "openEuler-SA-2024-1349",
+					FixedVersion:    "5.10.0-153.48.0.126",
+				},
+			},
+		},
+		{
+			name:     "happy path 2",
+			fixtures: []string{"testdata/fixtures/happy.yaml"},
+			version:  "22.03-LTS-SP2",
+			pkgName:  "perf-debuginfo",
+			want: []types.Advisory{
+				{
+					VulnerabilityID: "openEuler-SA-2024-1349",
+					FixedVersion:    "5.10.0-153.48.0.126",
+				},
+			},
+		},
```

Comment:
I think we can leave one test

Reply:
Thank you! I have removed `happy path 2`.